sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfigBuilder.java

```diff
@@ -53,6 +61,43 @@ public CosmosContainerProactiveInitConfigBuilder setProactiveConnectionRegionsCo
         return this;
     }
 
+    /**
+     * Sets the time window represented as a {@link Duration} within which connections will be opened
+     * aggressively and in a blocking manner and outside which connections will be opened defensively
+     * and in a non-blocking manner
+     *
+     * @param aggressiveWarmupDuration this denotes the aggressive proactive connection
+     *                                                             establishment duration to be set
+     * @return current {@link CosmosContainerProactiveInitConfigBuilder}
+     */
+    public CosmosContainerProactiveInitConfigBuilder setAggressiveWarmupDuration(Duration aggressiveWarmupDuration) {
+        checkArgument(aggressiveWarmupDuration != null,
+                "aggressiveWarmupDuration cannot be a null value");
+        checkArgument(aggressiveWarmupDuration.compareTo(Duration.ZERO) > 0,
+                "aggressiveWarmupDuration should be greater than Duration.ZERO");
+        this.aggressiveWarmupDuration = aggressiveWarmupDuration;
+        return this;
+    }
+
+    /**
+     * Sets the minimum no. of connections required to be opened for each replica of
+     * the container specified in the {@link CosmosContainerIdentity} instance.
+     *
+     * @param cosmosContainerIdentity Encapsulates the identity for the container for which minimum no. of connections
+     *                                are to be opened.
+     * @param minConnectionsPerEndpoint Denotes the minimum no. of connections to be opened for each endpoint of the container
+     *
+     * @return Current {@link CosmosContainerProactiveInitConfigBuilder}
+     * */
+    CosmosContainerProactiveInitConfigBuilder withMinConnectionsPerEndpointForContainer(CosmosContainerIdentity cosmosContainerIdentity, int minConnectionsPerEndpoint) {
```

Comment:
Please add a safe guard here - otherwise we will see CRIs because cx uses int.maxValue etc.

Reply:
For example minConnectionPerEndpoint has to be <= 10 or even <= 5